Scanning quits when trying to download from invalid URL

When I scan the results of the analyzer, the scanning process will quit when sources for a dependency cannot be downloaded due to an invalid URL.

This is the stacktrace:

````
Exception in thread "main" org.eclipse.jgit.api.errors.JGitInternalException: Exception caught during execution of ls-remote command
        at org.eclipse.jgit.api.LsRemoteCommand.execute(LsRemoteCommand.java:218)
        at org.eclipse.jgit.api.LsRemoteCommand.call(LsRemoteCommand.java:161)
        at com.here.ort.downloader.vcs.GitWorkingTree.listRemoteTags(GitWorkingTree.kt:116)
        at com.here.ort.downloader.WorkingTree.guessRevisionName(WorkingTree.kt:89)
        at com.here.ort.downloader.VersionControlSystem.download(VersionControlSystem.kt:261)
        at com.here.ort.downloader.VersionControlSystem.download$default(VersionControlSystem.kt:226)
        at com.here.ort.downloader.Downloader.downloadFromVcs(Downloader.kt:256)
        at com.here.ort.downloader.Downloader.download(Downloader.kt:144)
        at com.here.ort.downloader.Downloader.download$default(Downloader.kt:129)
        at com.here.ort.scanner.LocalScanner.scanPackage(LocalScanner.kt:287)
        at com.here.ort.scanner.LocalScanner.access$scanPackage(LocalScanner.kt:75)
        at com.here.ort.scanner.LocalScanner$scanPackages$2$invokeSuspend$$inlined$map$lambda$1$4.invokeSuspend(LocalScanner.kt:195)
        at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(ContinuationImpl.kt:33)
        at kotlinx.coroutines.DispatchedTask.run(DispatchedTask.kt:56)
        at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
        at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
        at java.base/java.lang.Thread.run(Thread.java:834)
Caused by: org.eclipse.jgit.errors.NotSupportedException: URI not supported: file://author-name@github.com/author-name/mylib.git
        at org.eclipse.jgit.transport.Transport.open(Transport.java:561)
        at org.eclipse.jgit.transport.Transport.open(Transport.java:429)
        at org.eclipse.jgit.transport.Transport.open(Transport.java:308)
        at org.eclipse.jgit.transport.Transport.open(Transport.java:276)
        at org.eclipse.jgit.api.LsRemoteCommand.execute(LsRemoteCommand.java:188)
````

As you can see, the analyser found a a VCS URL of the form `file://author-name@github.com/author-name/mylib.git` which cannot be used for downloading files. A `DownloadException` is not thrown in this case.

When I fix the URL to `https://github.com/author-name/mylib.git`, the scanner/downloader works as expected.

I have been advised to ping @sschuberth on this issue. ;-)

Thank you.
Uwe

--- a/downloader/src/main/kotlin/vcs/GitWorkingTree.kt
+++ b/downloader/src/main/kotlin/vcs/GitWorkingTree.kt
@@ -24,15 +24,14 @@ import com.here.ort.model.VcsInfo
 import com.here.ort.utils.log
 
 import java.io.File
+import java.io.IOException
 
 import org.eclipse.jgit.api.LsRemoteCommand
-import org.eclipse.jgit.api.errors.GitAPIException
 import org.eclipse.jgit.lib.BranchConfig
 import org.eclipse.jgit.lib.Constants
 import org.eclipse.jgit.lib.Repository
 import org.eclipse.jgit.storage.file.FileRepositoryBuilder
 import org.eclipse.jgit.submodule.SubmoduleWalk
-import java.io.IOException
 
 private fun findGitOrSubmoduleDir(workingDir: File): Repository {
     // First try to open an existing working tree exactly at the given directory. This also works for submodules which
@@ -91,7 +90,7 @@ open class GitWorkingTree(workingDir: File, private val gitBase: GitBase) : Work
         listSubmodulePaths(repo).associateWith { GitWorkingTree(repo.workTree.resolve(it), gitBase).getInfo() }
 
     override fun getRemoteUrl(): String =
-        try {
+        runCatching {
             val remotes = org.eclipse.jgit.api.Git(repo).remoteList().call()
             val remoteForCurrentBranch = BranchConfig(repo.config, repo.branch).remote
 
@@ -104,8 +103,8 @@ open class GitWorkingTree(workingDir: File, private val gitBase: GitBase) : Work
             }
 
             remote?.urIs?.firstOrNull()?.toString().orEmpty()
-        } catch (e: GitAPIException) {
-            throw IOException("Unable to get the remote URL.", e)
+        }.getOrElse {
+            throw IOException("Unable to get the remote URL.", it)
         }
 
     override fun getRevision(): String = repo.exactRef(Constants.HEAD)?.objectId?.name().orEmpty()
@@ -113,16 +112,16 @@ open class GitWorkingTree(workingDir: File, private val gitBase: GitBase) : Work
     override fun getRootPath(): File = repo.workTree ?: workingDir
 
     override fun listRemoteBranches(): List<String> =
-        try {
+        runCatching {
             LsRemoteCommand(repo).setHeads(true).call().map { it.name.removePrefix("refs/heads/") }
-        } catch (e: GitAPIException) {
-            throw IOException("Unable to list the remote branches.", e)
+        }.getOrElse {
+            throw IOException("Unable to list the remote branches.", it)
         }
 
     override fun listRemoteTags(): List<String> =
-        try {
+        runCatching {
             LsRemoteCommand(repo).setTags(true).call().map { it.name.removePrefix("refs/tags/") }
-        } catch (e: GitAPIException) {
-            throw IOException("Unable to list the remote tags.", e)
+        }.getOrElse {
+            throw IOException("Unable to list the remote tags.", it)
         }
 }

